dubbo-common/src/main/java/org/apache/dubbo/common/beans/factory/ScopeBeanFactory.java

```diff
@@ -124,19 +110,16 @@ public <T> T getOrRegisterBean(Class<T> type) {
         return getOrRegisterBean(null, type);
     }
 
-    public <T> T getOrRegisterBean(String name, Class<T> type) {
-        T bean = getBean(name, type);
-        if (bean == null) {
-            // lock by type
-            synchronized (type) {
-                bean = getBean(name, type);
-                if (bean == null) {
-                    bean = createAndRegisterBean(name, type);
-                }
+    private static final Hashtable<Class<?>, ReentrantLock> typeLocks = new Hashtable<>();
```

Comment:
Why not use `ConcurrentHashMap`?

Reply:
new commit: replace hashtable with concurrenthashmap.